diffkemp/semdiff/function_diff.py

```diff
@@ -124,9 +126,117 @@ def functions_semdiff(first, second, fun_first, fun_second, config):
         return result
 
 
+def _edge_callstack_to_string(callstack):
+    """Converts a callstack consisting of Edge objects to a string
+    representation."""
+    return "\n".join(["{} at {}:{}".format(call.target_name,
+                                           call.filename,
+                                           call.line)
+                      for call in callstack])
+
+
+def graph_to_fun_pair_list(fun_first, fun_second, graph):
+    # Extract the functions that should be compared from the graph in
+    # the form of Vertex objects.
+    vertices_to_compare = graph.reachable_from(
+        ComparisonGraph.Side.LEFT, fun_first)
+    # Use methods from ComparisonGraph (on the graph variable) and
+    # vertices_to_compare to generate objects_to_compare.
+    objects_to_compare = []
+    syndiff_bodies_left = dict()
+    syndiff_bodies_right = dict()
+    for vertex in vertices_to_compare:
+        if vertex.result in [Result.Kind.EQUAL,
+                             Result.Kind.ASSUMED_EQUAL]:
+            # Do not include equal functions into the result.
+            continue
+        # Generate and add the function difference.
+        fun_pair = []
+        for side in [ComparisonGraph.Side.LEFT,
+                     ComparisonGraph.Side.RIGHT]:
```

Comment:
This appears multiple times, could be replaced by `for side in ComparisonGraph.sides()`

Reply:
Actually just `for side in ComparisonGraph.Side` works (and is already used in `normalize`).